### PR TITLE
CMX: ingress work

### PR DESCRIPTION
**Pull request by marccampbell**
additional work on cmx ingress functionality

### CODE REVIEW

File: cli/cmd/cluster_addon_ingress_create.go [hunk 1]
```diff
@@ -13,28 +13,33 @@ import (
...
+	namespace := r.args.clusterCreateIngressNamespace
+	if namespace == "" {
+		namespace = "default" // avoiding the entire k8s dep list
+	}
 
 	opts := kotsclient.CreateClusterIngressOpts{
 		ClusterID: clusterID,
-		Target:    ingressTarget,
+		Target:    r.args.clusterCreateIngressTarget,
+		Namespace: namespace,
```
Comment (jdewinne) [inline cli/cmd/cluster_addon_ingress_create.go:42]: Port?

File: pkg/kotsclient/cluster_ingress_create.go [hunk 2]
```diff
@@ -9,10 +9,12 @@ import (
 type CreateClusterIngressOpts struct {
 	ClusterID string
 	Target    string
+	Namespace string
```
Comment (jdewinne) [inline pkg/kotsclient/cluster_ingress_create.go:12]: Port?

File: pkg/kotsclient/cluster_ingress_create.go [hunk 3]
```diff
@@ -9,10 +9,12 @@ import (
...
 }
 
 type CreateClusterIngressRequest struct {
-	Target string `json:"target"`
+	Target    string `json:"target"`
+	Namespace string `json:"namespace"`
```
Comment (jdewinne) [inline pkg/kotsclient/cluster_ingress_create.go:17]: Port?

File: pkg/kotsclient/cluster_ingress_create.go [hunk 4]
```diff
@@ -21,7 +23,8 @@ type CreateClusterIngressResponse struct {
 
 func (c *VendorV3Client) CreateClusterIngress(opts CreateClusterIngressOpts) (*types.ClusterAddOn, error) {
 	req := CreateClusterIngressRequest{
-		Target: opts.Target,
+		Target:    opts.Target,
+		Namespace: opts.Namespace,
```
Comment (jdewinne) [inline pkg/kotsclient/cluster_ingress_create.go:27]: Port?

File: pkg/types/cluster.go [hunk 5]
```diff
@@ -65,6 +65,7 @@ type ClusterAddOn struct {
 }
 
 type ClusterIngressAddOn struct {
-	Hostname string `json:"hostname"`
-	Target   string `json:"target"`
+	Hostname  string `json:"hostname"`
+	Target    string `json:"target"`
+	Namespace string `json:"namespace"`
```
Comment (jdewinne) [inline pkg/types/cluster.go:70]: Port?

**Review by jdewinne — COMMENTED** [review]
I think you might want to add Port also?
Or remove it from the flags for now?

File: cli/cmd/cluster_addon_ingress_create.go [hunk 6]
```diff
@@ -13,28 +13,34 @@ import (
 
 func (r *runners) InitClusterAddOnIngressCreate(parent *cobra.Command) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:  "create",
+		Use:  "create <cluster_id> --target svc/my-service --port <port>",
```
Comment (divolgin) [inline cli/cmd/cluster_addon_ingress_create.go:16]: are target and port required?

**Review by divolgin — DISMISSED** [review]

**Review by divolgin — APPROVED** [review]

**Pull request merged**